### ISSUE TITLE
Parsing incito response

### ISSUE THREAD

**Issue by VitaliyShuba** [shopgun/shopgun-android-sdk#32]
Hello.
I am referring to this question I asked some time ago:
https://github.com/shopgun/shopgun-android-sdk/issues/25

Again, thank you for help, it works well.
But I have a questions about more detailed parsing of incito publication response on Android. 
Sorry if I write these questions in wrong repository, not sure which is better.

1) Can we use "accessibility_label" to get the price of the offer? 
For example: "accessibility_label":"Avocado, 1 bk, 26 kr. 
Is it safe to assume that last part after comma is the price?

2) How to get images of products if offer has more than 1 product? If we use "image" field from the object in "products" array - it returns bad gateway error. Do we need to provide any api key? But if we dig deeper there is AbsoluteLayout with childs ImageViews - which have correct image url for each product.
**Is it safe to assume this AbsoluteLayout will be always first child of the offer, and is it safe to map array of those images to array of offer products?**

**Comment by pkebbesen** [comment]
Hi @VitaliyShuba , can I get you to contact me (Peter) via hello@tjek.com please? Just to clarify what you are trying to achieve with Incito. Thx :)

**Comment by VitaliyShuba** [comment]
Hello @pkebbesen, just sent an email.

**Comment by AgneseBussone** [comment]
Hello @VitaliyShuba , you cannot make these assumptions on the incito structure.

We have an additional api to get details related to an offer in incito: 
`https://squid-api.tjek-staging.com/v4/rpc/get_offer_from_incito_publication_view`
and the setup is similar to what you already have for the incito (same headers explained here #25 )
Take a look at the output it provides: it contains information regarding the offer like title, description, images, price, etc. Probably all you need is there.
As an example, I can give you the data class we have for the response:
```
@Keep
@JsonClass(generateAdapter = true)
data class Offer(
        val id: String = "",
        val name: String = "",
        val description: String? = "",
        val images: List<Image> = emptyList(),
        @Json(name = "webshop_link")
        val webshop: String? = "",
        val price: Float = 0F,
        @Json(name = "currency_code")
        val currency: String = "",
        val savings: Float? = 0F,
        @Json(name = "piece_count")
        val pieceCount: PieceCount = PieceCount(),
        @Json(name = "unit_symbol")
        val unitSymbol: QuantityUnit = QuantityUnit.Piece,
        @Json(name = "unit_size")
        val unitSize: UnitSize = UnitSize(),
        val validity: Validity = Validity(),
        @Json(name = "visible_from")
        val visibleFrom: Date = Date(),
        val business: Business = Business())
```

You need to provide 2 inputs to the api, the `viewId` and the `publicationId`. You can use the javascript interface `viewClicked`.
In the issue #25 , towards the end, it's explained how to use it. I'll re-post it here quickly:
```
@JavascriptInterface
fun viewClicked(viewIds: Array<String>?) {
    if (viewIds == null) {
        return
    }
   for (id in viewIds) {
    val incitoOffer = getOfferFromMap(id, catalog?.incitoId ?: "")
    if (incitoOffer != null) {
        saveFeatureLabels(incitoOffer.featureLabels)
        
       // Here you can fire the network request for the offer

        break
    }
}
```
Look at #25 to see how the `IncitoOffer` type is defined (it's created during parsing and it contains the `viewId` and `publicationId` you need) and `getOfferFromMap()`.


I hope this helps 🙂. Let us know if you need any further information.

**Comment by VitaliyShuba** [comment]
Thank you. But I don't see the products in the offer data class. So I assume it is safe to use the products from the original incito publication json?

**Comment by VitaliyShuba** [comment]
Yes, I am referring to offers that contain multiple products. In the incito publication object which I receive from this request
https://squid-api.tjek.com/v4/rpc/generate_incito_from_publication
I can easy parse all products of the offer, but with wrong links to the images.

<img width="535" alt="image" src="https://user-images.githubusercontent.com/4703948/154513008-e376db86-f704-49bb-9e86-0bf53fd278f0.png">

**Comment by pkebbesen** [comment]
@martensievers Could you help here please?

**Comment by martensievers** [comment]
Hi @VitaliyShuba, I removed your latest comment as the screenshot contained sensitive data. Now to your question, there is, unfortunately, no way right now that you can retrieve all the individual product images for an offer from within the Incito. 

We are currently working on an update to our API so that you will soon be able to retrieve the individual products of an offer via https://squid-api.tjek.com/v4/rpc/get_offer_from_incito_publication_view but that is not publicly available yet.

However, I think I may have a solution for you that I was made aware of from one of your colleagues today so I have sent you an email with further details on the matter.
